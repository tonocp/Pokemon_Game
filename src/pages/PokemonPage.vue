<template>
  <h1 v-if="!pokemon">Espere por favor...</h1>
  <div v-else>
    <h1>¿Quién es este pokémon?</h1>

    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />

    <PokemonOptions :pokemons="pokemonArr" />
  </div>
</template>

<script>
import PokemonPicture from "@/components/PokemonPicture";
import PokemonOptions from "@/components/PokemonOptions";
import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
    };
  },
  components: {
    PokemonPicture,
    PokemonOptions,
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonArr[rndInt];
    },
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>
