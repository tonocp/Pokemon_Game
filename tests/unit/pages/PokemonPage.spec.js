import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from "@/pages/PokemonPage";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonPage Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("Debe hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot;
  });

  test("Debe llamar mixPokemonArray al montar", () => {
    const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, "mixPokemonArray");
    shallowMount(PokemonPage);

    expect(mixPokemonArraySpy).toHaveBeenCalled();
  });

  test("Debe hacer match con el snapshot cuando cargan los pokemons", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
          counter: 0,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Debe de mostrar los componentes de PokemonPicture y PokemonOptions", () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
          counter: 0,
        };
      },
    });
    const picture = wrapper.find("pokemon-picture-stub");
    const options = wrapper.find("pokemon-options-stub");

    expect(picture.exists()).toBeTruthy();
    expect(picture.attributes("pokemonid")).toBe("1");

    expect(options.exists()).toBeTruthy();
    expect(options.attributes("pokemons")).toBeTruthy();
  });

  test("Pruebas con checkAnswer()", async () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
          counter: 0,
        };
      },
    });

    await wrapper.vm.checkAnswer(1);

    expect(wrapper.find("h3").exists()).toBeTruthy();
    expect(wrapper.vm.showPokemon).toBeTruthy();
    expect(wrapper.find("h3").text()).toBe(`Corecto! era ${pokemons[0].name} !`);

    await wrapper.vm.checkAnswer(3);
    expect(wrapper.vm.message).toBe(`Oops!, el correcto era ${pokemons[0].name} !`);
  });
});
