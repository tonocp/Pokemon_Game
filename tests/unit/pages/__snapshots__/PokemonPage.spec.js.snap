// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Debe hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es este pokémon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false" style="margin-top: 25px;"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]" style="margin-top: 25px;"></pokemon-options-stub>
  <h2>Puntuación = 0</h2>
  <!--v-if-->
</div>
`;
